sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -711,12 +711,15 @@ class SparkSession private(
   // scalastyle:on
 
   /**
-   * Stop the underlying `SparkContext`.
+   * Stop the underlying `SparkContext` if there are are no active sessions remaining.
    *
    * @since 2.0.0
    */
   def stop(): Unit = {
```

Comment:
Hey, I think this was a design decision that stopping sessions stops spark context too. Why don't you just don't call `stop()` on the session since what it does it just stops the session? Seems like the behaviour is documented properly as well.

Reply:
The idea is that if one creates a multi-tenant Spark process, and you give each user a Spark session, you want to be able to close down the resources for one session (e.g. connections to JDBC, perhaps), but not stop the entire Spark Context, thus keeping the Spark Context alive for the other users.